src/seed_tools/commands/create_seed.ts

```diff
@@ -13,69 +13,165 @@ import {
-  serial_number_path?: string;
+  output_serial_number_file?: string;
+  validate_only?: true;
   version?: string;
 }
 
-async function main(studiesDir: string, outputFile: string, options: Options) {
-  const files: string[] = [];
-  for (const dirItem of await fs.readdir(studiesDir, { withFileTypes: true })) {
-    if (!dirItem.isFile()) {
-      continue;
-    }
-    files.push(dirItem.name);
+async function main(
+  studiesDir: string,
+  outputSeedFile: string | undefined,
+  options: Options,
+) {
+  console.log(outputSeedFile, options);
+  if (outputSeedFile === undefined && !options.validate_only) {
+    console.error(
+      'Either output_seed_file or --validate_only must be provided',
+    );
+    process.exit(1);
   }
-  files.sort();
+
+  if (options.fix && outputSeedFile) {
+    console.log(`outputSeedFile: "${outputSeedFile}"`);
+    console.error(
+      'The --fix option cannot be used when an output_seed_file is provided',
+    );
+    process.exit(1);
+  }
+
+  const { studies, studyFileBaseNameMap, errors } =
+    await readStudiesFromDirectory(studiesDir, options);
 
   const variationsSeed: VariationsSeed = {
-    study: [],
+    study: studies,
     layers: [],
     version: options.version ?? '1',
   };
 
+  const serialNumber =
+    options.mock_serial_number ?? generateSerialNumber(variationsSeed);
+  variationsSeed.serial_number = serialNumber;
+
+  errors.push(
+    ...seed_validation.getSeedErrors(variationsSeed, studyFileBaseNameMap),
+  );
+
+  if (errors.length > 0) {
+    console.error(`Seed validation errors:\n${errors.join('\n---\n')}`);
+    process.exit(1);
+  }
+
+  console.log('Seed study count:', variationsSeed.study.length);
+  if (outputSeedFile !== undefined) {
+    const seedBinary = VariationsSeed.toBinary(variationsSeed);
+    await fs.writeFile(outputSeedFile, seedBinary);
+    if (options.output_serial_number_file !== undefined) {
+      await fs.writeFile(options.output_serial_number_file, serialNumber);
+    }
+    console.log(outputSeedFile, 'created with serial number', serialNumber);
+  }
+}
+
+async function readStudiesFromDirectory(
+  studiesDir: string,
+  options: Options,
+): Promise<{
+  studies: Study[];
+  studyFileBaseNameMap: Map<Study, string>;
+  errors: string[];
+}> {
+  const files = (await fs.readdir(studiesDir)).sort();
+
+  const studies: Study[] = [];
+  const studyFileBaseNameMap = new Map<Study, string>();
+  const errors: string[] = [];
+
   for (const file of files) {
     const filePath = path.join(studiesDir, file);
-    const studies = await study_json_utils.readStudyFile(filePath);
-    for (const study of studies) {
-      study_validation.validateStudy(study, filePath);
-      setStudyFixedParameters(study);
-      variationsSeed.study.push(study);
+    const readStudyFileResult = await study_json_utils.readStudyFile(filePath);
+    errors.push(...readStudyFileResult.errors);
+    errors.push(
+      ...(await checkAndOptionallyFixFormat(
```

Comment:
if there are `readStudyFileResult.errors` this code incorrectly suggests to "format" the file by replacing it to `[]` ;) 
```
Found unknown field while reading variations.Study.Experiment.Param from JSON format. JSON key: name1 (studies/BraveAdsTextClassificationPageProbabilitiesStudy.json)
---
Format required:
diff --git studies/BraveAdsTextClassificationPageProbabilitiesStudy.json studies/BraveAdsTextClassificationPageProbabilitiesStudy.json.formatted
index 5055002..fe51488 100644
--- studies/BraveAdsTextClassificationPageProbabilitiesStudy.json
+++ studies/BraveAdsTextClassificationPageProbabilitiesStudy.json.formatted
@@ -1,40 +1 @@
-[
-  {
-    "name": "BraveAdsTextClassificationPageProbabilitiesStudy",
...
-  }
-]
+[]
```

Reply:
thanks, fixed.